dotnet/src/SemanticKernel.Skills/Skills.Compression/FileCompressionSkill.cs

```diff
@@ -0,0 +1,125 @@
+//
+// ISemanticKernel kernel = SemanticKernel.Build();
+// var zipCompressor = new ZipFileCompressor();
+// var skill = new FileCompressionSkill(zipCompressor);
+// var fileCompression = kernel.ImportSkill(skill, "FileCompression");
+// string sourceFilePath = "FileToCompress.txt";
+// string destinationFilePath = "CompressedFile.zip";
+// var variables = new ContextVariables(sourceFilePath);
+// variables.Set(FileCompressionSkill.Parameters.DestinationFilePath, destinationFilePath);
+// await kernel.RunAsync(variables, fileCompression["CompressFileAsync"]);
+//**********************************************************************************************************************
+
+/// <summary>
+/// Skill for compressing and decompressing files.
+/// </summary>
+public class FileCompressionSkill
+{
+    /// <summary>
+    /// Parameter names.
+    /// <see cref="ContextVariables"/>
+    /// </summary>
+    public static class Parameters
+    {
+        /// <summary>
+        /// Directory to which to extract compressed file's data.
+        /// </summary>
+        public const string DestinationDirectoryPath = "destinationDirectoryPath";
+
+        /// <summary>
+        /// File path where to save compressed data.
+        /// </summary>
+        public const string DestinationFilePath = "destinationFilePath";
+    }
+
+    private readonly IFileCompressor _fileCompressor;
+    private readonly ILogger<FileCompressionSkill> _logger;
+
+    /// <summary>
+    /// Initializes a new instance of the <see cref="FileCompressionSkill"/> class.
+    /// </summary>
+    /// <param name="fileCompressor">File compressor implementation</param>
+    /// <param name="logger">Optional logger</param>
+    public FileCompressionSkill(IFileCompressor fileCompressor, ILogger<FileCompressionSkill>? logger = null)
+    {
+        this._fileCompressor = fileCompressor ?? throw new ArgumentNullException(nameof(fileCompressor));
+        this._logger = logger ?? new NullLogger<FileCompressionSkill>();
+    }
+
+    [SKFunction("Compresses an input file to an output file")]
+    [SKFunctionInput(Description = "Path of file to compress")]
+    [SKFunctionContextParameter(Name = Parameters.DestinationFilePath, Description = "Path of compressed file to create")]
+    public Task CompressFileAsync(string sourceFilePath, SKContext context)
+    {
+        this._logger.LogDebug($"{nameof(CompressFileAsync)} got called");
+
+        if (!context.Variables.Get(Parameters.DestinationFilePath, out string destinationFilePath))
+        {
+            this._logger.LogError($"Missing context variable in {nameof(CompressFileAsync)}");
+            string errorMessage = $"Missing variable {Parameters.DestinationFilePath}";
+            context.Fail(errorMessage);
+
+            return Task.FromException(new KeyNotFoundException(errorMessage));
+        }
+
+        return this._fileCompressor.CompressFileAsync(sourceFilePath, destinationFilePath, context.CancellationToken);
```

Comment:
Could we consider adding ```Environment.ExpandEnvironmentVariables``` to both the source and destination paths?

This exists as part of the DocumentSkill also and makes it easy for the caller to get access to a writable temp directory via %temp%: https://github.com/microsoft/semantic-kernel/blob/1ed81a4d3ac9f8e6c64d3ec26f7c0df05c5bbb17/dotnet/src/SemanticKernel.Skills/Skills.Document/FileSystem/LocalFileSystemConnector.cs#L31

Reply:
Done